src/queue.rs

```diff
@@ -6,22 +6,27 @@ use crate::tasklet::TaskletPtr;
 
 /// Trait for generic queue that stores data of type `T`.
 ///
+/// # Generic Parameters
 /// * `T` - Type of the stored data.
 pub(crate) trait Queue<T>: DataProvider<T> {
     /// Registers task to this queue.
     ///
+    /// # Parameters
     /// * `task` - Task to register.
     ///
-    /// Returns `InitError` in case of an error, `Ok(())` otherwise.
-    fn register_tasklet(&self, tasklet: TaskletPtr) -> Result<(), InitError>;
+    /// # Returns
+    /// `()` if successful, `InitError` otherwise.
+    unsafe fn register_tasklet(&self, tasklet: TaskletPtr) -> Result<(), InitError>;
```

Comment:
safety comment?

Reply:
```suggestion
    ///
    /// # Safety
    /// This is unsafe, because it mutably borrows the list of registered tasklets.
    /// This is safe to call before the system initialization.
    unsafe fn register_tasklet(&self, tasklet: TaskletPtr) -> Result<(), InitError>;
```